crates/bifrost/src/loglet/loglet_tests.rs

```diff
@@ -393,3 +399,150 @@ pub async fn single_loglet_readstream_test_with_trims(
+    for appender_id in 0..concurrent_appenders {
+        appenders.spawn({
+            let loglet = loglet.clone();
+            let append_barrier = append_barrier.clone();
+            async move {
+                let mut i = 1;
+                let mut committed = Vec::new();
+                let mut warmup = true;
+                loop {
+                    let res = loglet
+                        .append(Bytes::from(format!("appender-{}-record{}", appender_id, i)))
+                        .await;
+                    i += 1;
+                    if i > warmup_appends && warmup {
+                        println!("appender({}) - warmup complete....", appender_id);
+                        append_barrier.wait().await;
+                        warmup = false;
+                    }
+                    match res {
+                        Ok(offset) => {
+                            committed.push(offset);
+                        }
+                        Err(AppendError::Sealed) => {
+                            break;
+                        }
+                        Err(e) => fail!("unexpected error: {}", e)?,
+                    }
+                    // give a chance to other tasks to work
+                    tokio::task::yield_now().await;
+                }
+                Ok(committed)
+            }
+        });
+    }
+
+    // Wait for some warmup appends
+    println!(
+        "Awaiting all appenders to reach at least {} appends",
+        warmup_appends
+    );
+    append_barrier.wait().await;
+    // Go places and do other things.
+    for _ in 0..5 {
+        tokio::task::yield_now().await;
+    }
+
+    loglet.seal().await?;
+    // fails immediately
+    assert_that!(
+        loglet.append(Bytes::from_static(b"failed-record")).await,
+        err(pat!(AppendError::Sealed))
+    );
+
+    let tail = loglet.find_tail().await?;
+    assert!(tail.is_sealed());
+    println!("Sealed tail: {:?}", tail);
+
+    let mut all_committed = BTreeSet::new();
+    while let Some(handle) = appenders.join_next().await {
+        let mut committed = handle??;
+        assert!(!committed.is_empty());
+        let committed_len = committed.len();
+        assert!(committed_len >= 1000);
+        let tail_record = committed.pop().unwrap();
+        // tail must be beyond seal point
+        assert!(tail.offset() > tail_record);
+        println!(
+            "Committed len: {}, last appended was {}",
+            committed_len, tail_record
+        );
+        // ensure that all committed records are unique
+        assert!(all_committed.insert(tail_record));
+        for offset in committed {
+            assert!(all_committed.insert(offset));
+        }
+    }
+
+    let reader = loglet
+        .clone()
+        .create_read_stream(LogletOffset::OLDEST, Some(tail.offset().prev()))
+        .await?;
+
+    let records: BTreeSet<LogletOffset> = reader
+        .try_take_while(|x| std::future::ready(Ok(x.offset < tail.offset())))
```

Comment:
Is this an extra safety measure to not read beyond `tail.offset().prev()`?

Reply:
Yes. It's unnecessary, let me remove it.